ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
```

Comment:
Yeah. Most of these are split between an outwardly visible method and an impl method. Take a look: https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/722